File: src/Homework8/Homework8/Calculator/Calculator.cs
using Homework8.Interfaces;
using Homework8.Parser;

namespace Homework8.Calculator
{
    public class Calculator : ICalculator
    {
        public decimal Calculate(IInput input)
        {
            return input.Operation switch
            {
                SupportedOperations.Plus => input.Value1 + input.Value2,
                SupportedOperations.Minus => input.Value1 - input.Value2,
                SupportedOperations.Multiply => input.Value1 * input.Value2,
                SupportedOperations.Divide => input.Value1 / input.Value2,
            };
        }
    }
}

File: src/Homework4/Homework4/Calculator.fs
module Homework4.Calculator

type SupportedOperations =
    | Plus = 1
    | Minus = 2
    | Divide = 3
    | Multiply = 4 
let calculate val1 operation val2 =
        let tempVal1 = val1 |> double
        let tempVal2 = val2 |> double
        match operation with
        | SupportedOperations.Plus -> tempVal1 + tempVal2
        | SupportedOperations.Minus -> tempVal1 - tempVal2
        | SupportedOperations.Divide ->tempVal1 / tempVal2
        | _ -> tempVal1  * tempVal2